### ISSUE TITLE
react/prop-types: false positive

### ISSUE THREAD

**Issue by ferhatelmas** [yannickcr/eslint-plugin-react#1998]
After babel-eslint 10.0

```js
type Props = {
  onRowSelected: string => void,
  users: Array<UserType>
}

export default class NetworkTable extends Component<Props> {
  onRowSelected = (event: { data: { profile: string } }) => {
    const { onRowSelected } = this.props
    onRowSelected(event.data.profile)
  }
```
```sh
.../table.js
  24:13  error  'onRowSelected' is missing in props validation  react/prop-types
```

Versions:
* eslint: 5.6.0
* eslint-plugin-react: 7.11.1
* babel-eslint: 10.0.0
* react: 16.5.2

**Comment by ljharb** [comment]
If you name the class field something else, does the error disappear?

**Comment by ferhatelmas** [comment]
@ljharb it doesn't disappear but it seems https://github.com/babel/babel-eslint/pull/697 fixed issue for us.

**Comment by ljharb** [comment]
OK - I'm going to close this then, as it seems a bug in babel-eslint.
